abi_test.go

```diff
@@ -132,6 +159,53 @@ func TestProgramABI(t *testing.T) {
 	}
 }
 
+func TestNewProgramABIFromProc(t *testing.T) {
+	prog := createSocketFilter(t)
+	defer prog.Close()
+
+	name, abi, err := newProgramABIFromProc(prog.fd)
+	if err != nil {
+		t.Fatal("Can't read ABI:", err)
+	}
+
+	// This is the hash of the eBPF bytecode. It changes
+	// if createSocketFilter is changed.
+	if name != "d7edec644f05498d" {
```

Comment:
Who provides the original of this hash? I'm wondering if it's fixed in stone or whether the kernel may decide to change it in some future version.

Reply:
I think it's the truncated sha1 hash of the BPF bytecode, and you're right that the kernel may change this. I could change the test to be `name == ""`?